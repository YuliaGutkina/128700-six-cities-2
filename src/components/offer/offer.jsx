import React, {PureComponent} from 'react';
import {withRouter} from "react-router-dom";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import classNames from "classnames";

import Header from "../header/header";
import {
  receiveNearbyOffersSelector, receiveOfferCommentsSelector,
  receiveOfferSelector
} from "../../reducer/data/selectors";
import Bookmark from "../bookmark/bookmark";
import {ReviewsList} from "../reviews-list/reviews-list";
import {CityMap} from "../city-map/city-map";
import {OffersList} from "../offers-list/offers-list";
import {Operation} from "../../reducer/data/data";
import {Rating} from "../rating/rating";
import withReviewFormSubmit from "../../hocs/with-review-form-submit/with-review-form-submit";
import {ReviewForm} from "../review-form/review-form";
import {receiveUserDataSelector} from "../../reducer/user/selectors";


const ReviewFormWrapped = withReviewFormSubmit(ReviewForm);

class Offer extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {offerData, userData, nearbyOffers, comments = []} = this.props;

    return <div className="page">
      <Header/>
      {offerData &&
        <main className="page__main page__main--property">
          <section className="property">
            <div className="property__gallery-container container">
              <div className="property__gallery">
                {offerData.images.slice(0, 6).map((img) =>
                  <div
                    key={img}
                    className="property__image-wrapper"
                  >
                    <img className="property__image" src={img} alt="Photo studio" />
                  </div>
                )}
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                {offerData.isPremium &&
                  <div className="property__mark">
                    <span>Premium</span>
                  </div>}
                <div className="property__name-wrapper">
                  <h1 className="property__name">
                    {offerData.title}
                  </h1>
                  <Bookmark
                    offer={offerData}
                    className="property__bookmark-button"
                    iconClassName="property__bookmark-icon"
                    isLarge
                  />
                </div>
                <Rating
                  className="property__rating"
                  starsClassName="property__stars"
                  value={offerData.rating}
                />
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {offerData.type}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {offerData.bedrooms} Bedrooms
                  </li>
                  <li className="property__feature property__feature--adults">
                    Max {offerData.maxAdults} adults
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">€{offerData.price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <div className="property__inside">
                  <h2 className="property__inside-title">What&apos;s inside</h2>
                  <ul className="property__inside-list">
                    {offerData.goods.map((item) =>
                      <li key={item} className="property__inside-item">
                        {item}
                      </li>
                    )}
                  </ul>
                </div>
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <div
                      className={classNames(
                          `user__avatar-wrapper`,
                          `property__avatar-wrapper`,
                          {
                            "property__avatar-wrapper--pro": offerData.host.isPro
                          }
                      )}
                    >
                      <img className="property__avatar user__avatar" src={`/${offerData.host.avatar}`} width={74} height={74} alt="Host avatar" />
                    </div>
                    <span className="property__user-name">
                      {offerData.host.name}
                    </span>
                    {offerData.host.isPro &&
                    <span className="property__user-status">
                      Pro
                    </span>}
                  </div>
                  <div className="property__description">
                    <p className="property__text">
                      {offerData.description}
                    </p>
                  </div>
                </div>
                <section className="property__reviews reviews">
                  <h2 className="reviews__title">Reviews · <span className="reviews__amount">{comments.length}</span></h2>
                  <ReviewsList comments={comments.slice(0, 10)}/>
                  {userData &&
                    <ReviewFormWrapped
                      offerId={offerData.id}
                    />
                  }
                </section>
              </div>
            </div>
            <CityMap
              className="property__map"
              items={[offerData, ...nearbyOffers]}
              initialLocation={offerData.location}
              activeItem={offerData}
            />
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">Other places in the neighbourhood</h2>
              <OffersList
                className="near-places__list"
                places={nearbyOffers}
                cardClassName="near-places__card"
                imageWrapperClassName="near-places__image-wrapper"
                imageWidth={260}
                imageHeight={200}
              />
            </section>
          </div>
        </main>
      }
    </div>;
  }

  componentDidMount() {
    const {onLoadComments, match} = this.props;

    onLoadComments(match.params.id);
  }
}

Offer.propTypes = {
  userData: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
    name: PropTypes.string,
    avatar: PropTypes.string,
    isPro: PropTypes.bool
  }),
  offerData: PropTypes.object,
  nearbyOffers: PropTypes.array,
  onLoadComments: PropTypes.func,
  match: PropTypes.object,
  comments: PropTypes.array
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  userData: receiveUserDataSelector(state),
  offerData: receiveOfferSelector(state, ownProps.match.params.id),
  nearbyOffers: receiveNearbyOffersSelector(state, ownProps.match.params.id).slice(0, 3),
  comments: receiveOfferCommentsSelector(state, ownProps.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
  onLoadComments: (offerId) => {
    dispatch(Operation.loadComments(offerId));
  },
});


export {Offer};
export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Offer));

