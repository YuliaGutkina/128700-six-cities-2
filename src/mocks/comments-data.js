export const commentsData = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];
