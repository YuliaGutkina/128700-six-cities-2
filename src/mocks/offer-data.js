export const offersData = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 3,
    city: {
      name: `Tel Aviv`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 5,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 6,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 7,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 8,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 9,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  }
];
