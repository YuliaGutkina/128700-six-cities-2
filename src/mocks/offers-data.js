export const offersData = [
  {
    city: `Amsterdam`,
    initialCoordinates: [52.38333, 4.9],
    places: [
      {
        title: `Beautiful & luxurious apartment at great location`,
        type: `Apartment`,
        isPremium: true,
        src: `img/apartment-01.jpg`,
        price: 120,
        rating: 93,
        coordinates: [52.3909553943508, 4.85309666406198]
      },
      {
        title: `Wood and stone place`,
        type: `Private room`,
        isPremium: false,
        src: `img/room.jpg`,
        price: 80,
        rating: 80,
        coordinates: [52.369553943508, 4.85309666406198]
      },
      {
        title: `Canal View Prinsengracht`,
        type: `Apartment`,
        isPremium: false,
        src: `img/apartment-02.jpg`,
        price: 132,
        rating: 80,
        coordinates: [52.3909553943508, 4.929309666406198]
      },
      {
        title: `Nice, cozy, warm big bed apartment`,
        type: `Apartment`,
        isPremium: true,
        src: `img/apartment-03.jpg`,
        price: 180,
        rating: 100,
        coordinates: [52.3809553943508, 4.939309666406198]
      }
    ]
  },
  {
    city: `Paris`,
    initialCoordinates: [48.864716, 2.349014],
    places: []
  },
  {
    city: `Hamburg`,
    initialCoordinates: [53.551086, 9.993682],
    places: []
  }
];
