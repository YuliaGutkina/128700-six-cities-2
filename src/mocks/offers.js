export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    isPremium: true,
    src: `img/apartment-01.jpg`,
    price: 120,
    rating: 93
  },
  {
    title: `Wood and stone place`,
    type: `Private room`,
    isPremium: false,
    src: `img/room.jpg`,
    price: 80,
    rating: 80
  },
  {
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    isPremium: false,
    src: `img/apartment-02.jpg`,
    price: 132,
    rating: 80
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    isPremium: true,
    src: `img/apartment-03.jpg`,
    price: 180,
    rating: 100
  }
];
