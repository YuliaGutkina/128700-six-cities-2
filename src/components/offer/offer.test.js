import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import {offersData} from "../../mocks/offer-data";
import {Offer} from "./offer";
import {commentsData} from "../../mocks/comments-data";


it(`Favorites page correctly renders after relaunch`, () => {
  const renderer = new ShallowRenderer();
  renderer
    .render(
        <Offer
          offerData={offersData[0]}
          nearbyOffers={[offersData[1], offersData[2]]}
          onLoadComments={jest.fn()}
          comments={commentsData}
        />
    );
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
