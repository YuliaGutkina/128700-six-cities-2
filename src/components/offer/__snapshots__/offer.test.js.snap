// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites page correctly renders after relaunch 1`] = `
<div
  className="page"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/1.png"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/2.png"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious studio at great location
            </h1>
            <Memo(Connect(withRouter(Bookmark)))
              className="property__bookmark-button"
              iconClassName="property__bookmark-icon"
              isLarge={true}
              offer={
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.370216,
                      "longitude": 4.895168,
                      "zoom": 10,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "goods": Array [
                    "Heating",
                    "Kitchen",
                    "Cable TV",
                    "Washing machine",
                    "Coffee machine",
                    "Dishwasher",
                  ],
                  "host": Object {
                    "avatar": "img/1.png",
                    "id": 3,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "img/1.png",
                    "img/2.png",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 52.35514938496378,
                    "longitude": 4.673877537499948,
                    "zoom": 8,
                  },
                  "maxAdults": 4,
                  "preview": "img/1.png",
                  "price": 1200,
                  "rating": 4.8,
                  "title": "Beautiful & luxurious studio at great location",
                  "type": "apartment",
                }
              }
            />
          </div>
          <Rating
            className="property__rating"
            isValueVisible={true}
            starsClassName="property__stars"
            value={4.8}
            valueClassName="property__rating-value"
          />
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              1200
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Cable TV
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="user__avatar-wrapper property__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height={74}
                  src="/img/1.png"
                  width={74}
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                3
              </span>
            </h2>
            <ReviewsList
              comments={
                Array [
                  Object {
                    "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "date": "2019-05-08T14:13:56.569Z",
                    "id": 1,
                    "rating": 4,
                    "user": Object {
                      "avatar": "img/1.png",
                      "id": 4,
                      "isPro": false,
                      "name": "Max",
                    },
                  },
                  Object {
                    "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "date": "2019-05-08T14:13:56.569Z",
                    "id": 2,
                    "rating": 4,
                    "user": Object {
                      "avatar": "img/1.png",
                      "id": 4,
                      "isPro": false,
                      "name": "Max",
                    },
                  },
                  Object {
                    "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "date": "2019-05-08T14:13:56.569Z",
                    "id": 3,
                    "rating": 4,
                    "user": Object {
                      "avatar": "img/1.png",
                      "id": 4,
                      "isPro": false,
                      "name": "Max",
                    },
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
      <CityMap
        activeItem={
          Object {
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Amsterdam",
            },
            "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
            "goods": Array [
              "Heating",
              "Kitchen",
              "Cable TV",
              "Washing machine",
              "Coffee machine",
              "Dishwasher",
            ],
            "host": Object {
              "avatar": "img/1.png",
              "id": 3,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "img/1.png",
              "img/2.png",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 52.35514938496378,
              "longitude": 4.673877537499948,
              "zoom": 8,
            },
            "maxAdults": 4,
            "preview": "img/1.png",
            "price": 1200,
            "rating": 4.8,
            "title": "Beautiful & luxurious studio at great location",
            "type": "apartment",
          }
        }
        className="property__map"
        initialLocation={
          Object {
            "latitude": 52.35514938496378,
            "longitude": 4.673877537499948,
            "zoom": 8,
          }
        }
        items={
          Array [
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatar": "img/1.png",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "img/1.png",
                "img/2.png",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.35514938496378,
                "longitude": 4.673877537499948,
                "zoom": 8,
              },
              "maxAdults": 4,
              "preview": "img/1.png",
              "price": 1200,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "apartment",
            },
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatar": "img/1.png",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "img/1.png",
                "img/2.png",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.35514938496378,
                "longitude": 4.673877537499948,
                "zoom": 8,
              },
              "maxAdults": 4,
              "preview": "img/1.png",
              "price": 120,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "apartment",
            },
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Tel Aviv",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatar": "img/1.png",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "img/1.png",
                "img/2.png",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.35514938496378,
                "longitude": 4.673877537499948,
                "zoom": 8,
              },
              "maxAdults": 4,
              "preview": "img/1.png",
              "price": 120,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "apartment",
            },
          ]
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <OffersList
          cardClassName="near-places__card"
          className="near-places__list"
          imageHeight={200}
          imageWidth={260}
          imageWrapperClassName="near-places__image-wrapper"
          places={
            Array [
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar": "img/1.png",
                  "id": 3,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 2,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "maxAdults": 4,
                "preview": "img/1.png",
                "price": 120,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Tel Aviv",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar": "img/1.png",
                  "id": 3,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 3,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "maxAdults": 4,
                "preview": "img/1.png",
                "price": 120,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
            ]
          }
        />
      </section>
    </div>
  </main>
</div>
`;
